airflow/datasets/manager.py

```diff
@@ -0,0 +1,62 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from sqlalchemy.orm.session import Session
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class DatasetEventManager(LoggingMixin):
+    """
+    A pluggable class that manages operations for dataset events.
+
+    The intent is to have one place to handle all DatasetEvent-related operations, so different
+    Airflow deployments can use plugins that broadcast dataset events to each other.
+    """
+
+    def register_dataset_change(self, *, task_instance, dataset, extra=None, session: Session) -> None:
```

Comment:
We should do typing here for ti and dataset. I sorta think we can be picky too and just accept datasetmodel?

Reply:
Done.

Reply:
I tweaked it to only accept a `Dataset`, and put an isinstance check in the caller, since I _think_ task outlets can be things that aren't datasets.